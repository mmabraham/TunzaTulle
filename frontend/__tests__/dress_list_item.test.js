import React from 'react';
import { StaticRouter } from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import DressListItem from '../components/dresses/dress_list_item';
import renderer from 'react-test-renderer';

const dressData = {
    "id": 51,
    "barcode": "53245234523",
    "title": "this fakdfjlasdf",
    "color": "black and white",
    "description": "fadfgaerdheghdehteHTAEGTERG",
    "waist": "34.0",
    "min_waist": "32.5",
    "max_waist": "35.5",
    "sleeve_length": "above elbow",
    "height": "23.0",
    "min_height": "16.0",
    "max_height": "25.0",
    "age": 4,
    "min_age": 2,
    "max_age": 6,
    "img": "https://s3.amazonaws.com/tunza-dev/dresses/images/000/000/051/original/data?1505226569",
    "order_dates": []
  }

test('DressListItem renders correctly', () => {
  const component = renderer.create(
    <MuiThemeProvider>
      <StaticRouter location="/dresses" context={{}}>
        <DressListItem dress={dressData} />
      </StaticRouter>
    </MuiThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
