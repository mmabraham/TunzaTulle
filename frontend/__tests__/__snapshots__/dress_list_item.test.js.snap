// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DressListItem renders correctly 1`] = `
<a
  href="/dresses/51"
  onClick={[Function]}
>
  <div
    className="dress-list-tile"
    cols={1}
    rows={1}
    style={
      Object {
        "display": "block",
        "height": "100%",
        "muiPrepared": true,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <img
      src="https://s3.amazonaws.com/tunza-dev/dresses/images/000/000/051/original/data?1505226569"
      style={
        Object {
          "height": "100%",
          "left": "50%",
          "muiPrepared": true,
          "position": "relative",
          "transform": "translateX(-50%)",
        }
      }
    />
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "rgba(0, 0, 0, 0.4)",
          "bottom": 0,
          "display": "flex",
          "height": 68,
          "left": 0,
          "muiPrepared": true,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <div
        style={
          Object {
            "color": "#ffffff",
            "flexGrow": 1,
            "marginLeft": 16,
            "marginRight": 0,
            "muiPrepared": true,
            "overflow": "hidden",
          }
        }
      >
        <div
          style={
            Object {
              "fontSize": "16px",
              "muiPrepared": true,
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          this fakdfjlasdf
        </div>
        <div
          style={
            Object {
              "fontSize": "12px",
              "muiPrepared": true,
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          <span
            className="subtitle"
          >
            <b>
              53245234523
            </b>
            <b>
              waist: 
              34.0
            </b>
            <b>
              height: 
              23.0
            </b>
            <b>
              age: 
              4
            </b>
          </span>
        </div>
      </div>
    </div>
  </div>
</a>
`;
